src/mock/ray/pubsub/publisher.h

```diff
@@ -84,30 +78,40 @@ class MockPublisherInterface : public PublisherInterface {
 namespace ray {
 namespace pubsub {
 
+instrumented_io_context __io_service;
```

Comment:
Double underscore prefixes are reserved for compilers. Ideally, we don't definite variables in headers, unless with `inline` or as a static local variable: https://google.github.io/styleguide/cppguide.html#Static_and_Global_Variables. 

Also IIUC files under `src/mock` are generated programmatically at the beginning. Should we keep it that way and put customized mocks in other locations? Then we would be able to re-run the tool to generate new mocks if needed. Otherwise it seems easy for mocks and actual interfaces to diverge.